src/transformer_util.ts

```diff
@@ -202,8 +202,10 @@ function emitMissingSyntheticCommentsAfterTypescriptTransform(context: ts.Transf
       // TypeScript ignores synthetic comments on reexport / import statements.
       const moduleName = extractModuleNameFromRequireVariableStatement(node);
       if (moduleName && fileContext.importOrReexportDeclarations) {
-        const index = importOrReexportsCount++;
-        const originalImportOrReexportNode = fileContext.importOrReexportDeclarations[index];
+        const index = (importOrReexportsCount[moduleName] || 0);
+        importOrReexportsCount[moduleName] = index + 1;
+        const originalImportOrReexportNode =
+            (fileContext.importOrReexportDeclarations[moduleName] || [])[index];
```

Comment:
I changed the implementation to merge the comments for all import/export statements of the same moduleName and added details why.